otlp/queue_manager.go

```diff
@@ -214,11 +217,17 @@ func NewQueueManager(logger log.Logger, cfg promconfig.QueueConfig, timeout time
 
 // Append queues a sample to be sent to the OpenTelemetry API.
 // Always returns nil.
-func (t *QueueManager) Append(ctx context.Context, hash uint64, sample *metricspb.Metric) error {
+func (t *QueueManager) Append(ctx context.Context, sample *metricspb.Metric) error {
 	t.queueLengthCounter.Add(ctx, 1)
+	t.samplesIn.incr(1)
+	shards := t.shards.shards
 
 	t.shardsMtx.RLock()
-	t.shards.enqueue(hash, sample)
+
+	shardIndex := t.rnd.Intn(len(shards))
```

Comment:
I'm curious if you have any thoughts on whether using random here may cause some shards to end up doing most of the work. If so, do we need to make a note to ensure load is shared across shards?

Reply:
I am satisfied with _expected uniform_ load distribution, which is maybe dodging your question. The reason for this PR is to remove a vestige that gets in the way of reorganizing the queue manager in other ways.

The problems we are seeing have to do with tail latency. If we hypothesize a case where one specific metric has a significantly slower response time, then the former code would end up with the main thread (reading WAL files) blocked a lot of the time due to one shard, and the other shards sending under-sized requests. The new code will block the main thread in the same way, but with (expected) larger requests, thus anticipating higher throughput with more work being done by more shards.

The next related improvement: I don't want to be queueing points for shards that are busy writing; then we will only place samples into shards that are busy assembling requests.